native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -485,14 +489,19 @@ static apr_status_t create_worker(proxy_server_conf *conf, proxy_balancer *balan
             }
             return APR_SUCCESS; /* Done Already existing */
         }
-        ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, server,
-                     "Created: can't reuse worker as it for %s cleaning...", url);
+        ap_log_error(APLOG_MARK, APLOG_ERR, 0, server, SAMEURL, url);
+        ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, server, "Created: can't reuse worker as it for %s cleaning...", url);
         if (!worker->opaque)
             worker->opaque = (proxy_cluster_helper *) apr_pcalloc(conf->pool,  sizeof(proxy_cluster_helper));
         if (!worker->opaque)
             return APR_EGENERAL;
         if (worker->cp->pool) {
             /* destroy and create a new one */
+            ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, server, "Reslist stats: worker->cp->res->ntotal:%d, worker->cp->res->nidle:%d", worker->cp->res->ntotal, worker->cp->res->nidle);
+            //MODCLUSTER-448
+            if(worker->cp->res->ntotal != worker->cp->res->nidle) {
```

Comment:
If the number of idle and total is not the same, we can't simply destroy the pool for this worker at this time. It will be caught during next iteration.   This whole code branch is a mere corner case: It takes place when one tries to connect two worker with the same host and port at the same time. If you just remove and add 1 worker, you won't get ntotal != nidle ever.